Jagati.cmake

```diff
@@ -1991,13 +1991,21 @@ macro(EmitExceptionSource)
     set(NamespaceSourceHeader "\
 #include \"DataTypes.h\"\n\
 #include \"MezzException.h\"\n\n\
+SAVE_WARNING_STATE\n\n\
```

Comment:
I am assuming this sets the conditions in which an exception will be thrown?

Reply:
Not Quite.

ALL CAPS is a convention indicating a preprocess or macro. This is an instruction to the compiler to save its current state for compiler warnings. It puts a copy of the current state on a stack of settings, this pairs with RESTORE_WARNING_STATE to pop on state of the top of that stack and undo changes since the most recent save.

Reply:
Here is some reading on compiler warnings: https://stackoverflow.com/questions/57842756/why-should-i-always-enable-compiler-warnings

And here is one of Jason Turner's great C++ weekly video where he discusses compiler warnings:  https://www.youtube.com/watch?v=IOo8gTDMFkM